main/tests/test_views.py

```diff
@@ -0,0 +1,128 @@
+        authorization = self.client.login(username='admin', password='12345678')
+        self.assertTrue(authorization)
+        response = self.client.get('/post/1/edit/')
+        # This time we'are logged but post is not exist
+        self.assertEqual(404, response.status_code)
+        Post.objects.create(author=self.admin, title='Test', text='superText', created_date=timezone.now())
+        response = self.client.get('/post/1/edit/')
+        self.assertEqual(200, response.status_code)
+
+    def test_drafts(self):
+        """Testing drafts view before and after login."""
+        response = self.client.get('/drafts/')
+        self.assertEqual(302, response.status_code)
+        authorization = self.client.login(username='admin', password='12345678')
+        self.assertTrue(authorization)
+        response = self.client.get('/drafts/')
+        self.assertEqual(200, response.status_code)
+
+    def test_publish_post(self):
+        """Testing publishing post before login and after."""
+        response = self.client.get('/post/1/publish/')
+        self.assertEqual(302, response.status_code)
+        authorization = self.client.login(username='admin', password='12345678')
+        self.assertTrue(authorization)
+        response = self.client.get('/post/1/publish/')
+        self.assertEqual(404, response.status_code)
+        Post.objects.create(author=self.admin, title='Test', text='superText', created_date=timezone.now())
+        response = self.client.get('/post/1/publish/', follow=True)
+        # The same situation as with new post test
+        self.assertEqual(response.redirect_chain[0][0], '/post/1/')
+
+    def test_delete_post(self):
+        """Testing deleting post before and after login."""
+        response = self.client.get('/post/1/remove/')
+        self.assertEqual(302, response.status_code)
+        authorization = self.client.login(username='admin', password='12345678')
+        self.assertTrue(authorization)
+        response = self.client.get('/post/1/remove/')
+        self.assertEqual(404, response.status_code)
+        Post.objects.create(author=self.admin, title='Test', text='superText', created_date=timezone.now())
+        response = self.client.get('/post/1/remove/', follow=True)
+        self.assertEqual('/', response.redirect_chain[0][0])
+
+    def test_add_comment(self):
+        """Add comment to post."""
+        self.post = Post.objects.create(author=self.admin, title='Test', text='superText', created_date=timezone.now())
+        resqonse = self.client.get('/post/1/comment/')
+        self.assertEqual(200, resqonse.status_code)
+        response = self.client.post('/post/1/comment/', {'author': self.admin, 'text': 'Super',
+                                                         'created_date': timezone.now(), 'is_approved': False},
+                                    follow=True)
+        self.assertEqual('/post/1/', response.redirect_chain[0][0])
```

Comment:
Better use [assertRedirects](https://docs.djangoproject.com/en/1.10/topics/testing/tools/#django.test.SimpleTestCase.assertRedirects) in this case